src/Foundatio.Aliyun/Storage/AliyunFileStorage.cs

```diff
@@ -0,0 +1,199 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Aliyun.OSS;
+using Foundatio.Extensions;
+
+namespace Foundatio.Storage {
+    public class AliyunFileStorage : IFileStorage {
+        private readonly string _bucketName;
+        private readonly OssClient _client;
+
+        public AliyunFileStorage(string connectionString, string bucketName = "storage") {
+            var account = AliyunConnectionString.Parse(connectionString);
+            _client = account.CreateClient();
+            _bucketName = bucketName;
+            if (!_client.DoesBucketExist(_bucketName)) _client.CreateBucket(_bucketName);
+        }
+
+        public void Dispose() { }
+
+        private bool IsNotFoundException(Exception ex) {
+            if (ex is AggregateException aggregateException) {

```

Comment:
Is there a specific spot you came across an aggregation Exception? Typically if we are awaiting we shouldn't have any (at least that is my understanding)

Reply:
This method only been called in ExistsAsync. When the target file is not exist, the OssClient.DoesObjectExist will throw AggregateException which include only one inner exception that type of WebException with the status code is HttpStatusCode.NotFound.
I know the design of the OssClient is not as expected, but we should replace it with our own implementation. As you said, if so we have to face to the expense of maintenance and initial work.